lms/djangoapps/django_comment_client/base/views.py

```diff
@@ -478,12 +525,56 @@ def un_pin_thread(request, course_id, thread_id):
 @login_required
 @permitted
 def follow_thread(request, course_id, thread_id):
+    """
+    given a course id and thread id, follow this thread
+    ajax only
+    """
     user = cc.User.from_django_user(request.user)
     thread = cc.Thread.find(thread_id)
     user.follow(thread)
+    # Feature Flag to check that notifications are enabled or not.
+    if settings.FEATURES.get("NOTIFICATIONS_ENABLED", False):
+        # only send notifications when the user
+        # who is following the thread is not the same
+        # who created the thread
+        action_user_id = request.user.id
+        original_poster_id = int(thread.user_id)
+        if not original_poster_id == action_user_id:
+            register_and_publish_notifications(
+                msg_type_name='open-edx.lms.discussions.thread-followed',
+                course_id=course_id,
+                original_poster_id=original_poster_id,
+                action_user_id=action_user_id,
+                action_username=request.user.username,
+                thread_title=thread.title,
+                link_to_thread=permalink(thread)
+            )
     return JsonResponse({})
 
 
+def register_and_publish_notifications(msg_type_name, course_id, original_poster_id,  # pylint: disable=invalid-name
```

Comment:
since we're not registering NotificationTypes any longer here, can we rename this to be - say - publish_discussion_notification()?